src/codegen/dyngen.rs

```diff
@@ -125,24 +136,50 @@ impl DynamicItems {
             assert_eq!(args.len(), args_identifiers.len());
         }
 
-        self.struct_members.push(quote! {
-            pub #ident: Result<unsafe extern #abi fn ( #( #args ),* ) #ret, ::libloading::Error>,
-        });
+        self.has_required |= is_required;
+
+        self.struct_members.push(
+            if is_required {
+                quote! {
+                    pub #ident: unsafe extern #abi fn ( #( #args),* ) #ret,
```

Comment:
Would be good to at least extract the `unsafe extern ...` tokens to its own local, maybe:

```rust
let signature = quote! { unsafe extern #abi fn ( #( #args), *) #ret };
let member = if is_required {
    signature
} else {
    quote! { Result<#signature, ::libloading::Error> }
};
self.struct_members.push(quote! {
    pub #ident: #member,
});
```

Or something like that, wdyt?

Reply:
That would be perfect! I'll make the change.